kube-client/src/lib.rs

```diff
@@ -123,6 +123,26 @@ pub use crate::core::{CustomResourceExt, Resource, ResourceExt};
 /// Re-exports from kube_core
 pub use kube_core as core;
 
+pub mod prelude {
+    //! A "prelude" for kube client crate. Reduces the number of duplicated imports.
+    //!
+    //! This prelude is similar to the standard library's prelude in that you'll
+    //! almost always want to import its entire contents, but unlike the
+    //! standard library's prelude you'll have to do so manually:
+    //!
+    //! ```
+    //! # #[allow(unused_imports)]
+    //! use kube::client::prelude::*;
+    //! ```
+    //!
+    //! The prelude may grow over time as additional items see ubiquitous use.
+
+    #[cfg(feature = "unstable-client")] pub use crate::client::scope::*;
```

Comment:
We I see people are importing traits in prelude using `pub use ExplicitScope as _;`  to presumably avoid warnings
https://github.com/tokio-rs/tracing/blob/master/tracing-subscriber/src/prelude.rs

ideally people should not need to opt-in to `unused_imports`

Reply:
Are you sure adding feature gates in a prelude is a good idea? People doing a wildcard import on prelude crate may get their symbols conditionally shadowed if another crate enabled this feature.

Reply:
I think the shadowing is a bigger problem than the feature gates, and the shadowing can be avoided by `use X as _;`. It limits the prelude to extension trait stuff, but maybe that is a better way forward?

Reply:
So as an update - I added all blanket implemented traits and extension traits to the prelude, while keeping those under feature gates separately - as prelude_unstable. I'm not sure that 2 preludes are needed, but went with separate so far. Also removed all struct, to avoid opt-in for unused imports. It would be perfect to have an option to say - prelude can be used but not re-exported (as in lib.rs), but I don't know if it is possible 🙂